Makefile

```diff
@@ -32,3 +32,36 @@ breakpad:
 
 json_enhancements_pg_extension: bootstrap
 	./scripts/json-enhancements.sh
+
+
+# Docker related rules
+
+.PHONY: dockerbuild dockersetup dockerclean dockertest dockerrun
+
+DC := $(shell which docker-compose)
+
+.docker-build:
+	make dockerbuild
+
+dockerbuild:
+	${DC} build processor
+	${DC} build webapp
+	${DC} build test
+	touch .docker-build
+
+# NOTE(willkg): We run setup in the webapp container because the webapp will own
+# postgres going forward and has the needed environment variables.
+# FIXME(willkg): These are not idempotent, but they should be. Plus we should
+# run migrations here, too.
+dockersetup: .docker-build
+	-${DC} run --entrypoint /app/docker/run_setup_postgres.sh webapp
+	-${DC} run --entrypoint /app/docker/run_setup_elasticsearch.sh webapp
+
+dockerclean:
+	rm .docker-build
+
+dockertest:
+	${DC} run test
+
+dockerrun:
+	${DC} up webapp processor
```

Comment:
I'm not clear on whether anyone will ever run this. I know I won't. Given that, it's probably prudent to nix this rule.

One of the things I tried to do was build the images and infrastructure in such a way that `docker-compose` subcommands were convenient and easy to use. Because of that, I think we can rely on make rules less.